docs/packages/python/symbol.md

```diff
@@ -38,6 +39,45 @@ The following code gives an example of computation graph that add two inputs tog
 >>> c = a + b
 ````
 
+Symbol Attributes
+-----------------
+Attributes can be attached to symbols, by providing an attribute dictionary when creating a symbol.
+```python
+data = mx.sym.Variable('data', attr={'mood': 'angry'})
+op   = mx.sym.Convolution(data=data, name='conv', kernel=(1, 1),
+                          num_filter=1, attr={'mood': 'so so'})
+```
+Both key and values of the attribute dictionary should be strings, in order to properly communicate with the C++ backend. The attributes can be retrived via `attr(key)` or `list_attr()`:
+```
+assert data.attr('mood') == 'angry'
+assert op.list_attr() == {'mood': 'so so'}
+```
+In the case of a composite symbol, you can also retrieve all the attributes associated with that symbol *and its descendents* via `list_attr_recursive()`. Note in the returned dictionary, all the attribute names are with a prefix `'symbol_name' + '/'` in order to avoid naming conflicts.
+```python
+assert op.list_attr_recursive() == {'data/mood': 'angry', 'conv/mood': 'so so',
```

Comment:
explain why mood is copied to conv_weight/mood here. also note that it won't happen if data is created by hand (is this true?) or if data is output of another symbol